File: android/app/src/main/kotlin/com/sotalog/android/domain/models/CallsignHistory.kt
package com.sotalog.android.domain.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

/**
 * Cached enrichment for a callsign (name/QTH/grid from QRZ and prior contacts).
 * The "times worked" count is NOT stored here — it is derived on demand from the
 * `qso` table (see [com.sotalog.android.data.local.database.dao.QSODao.countByCallsign]),
 * which is the single source of truth and stays correct across edits, deletes, and
 * imports automatically.
 */
@Entity(tableName = "callsignHistory")
data class CallsignHistory(
    @PrimaryKey
    val callsign: String,
    val name: String? = null,
    val qth: String? = null,
    val grid: String? = null,
    val lastWorked: Date? = null,
)

File: ios/SOTALog/Models/CallsignHistory.swift
import Foundation
import GRDB

/// Cached enrichment for a callsign (name/QTH/grid from QRZ and prior contacts).
/// The "times worked" count is NOT stored here — it is derived on demand from the
/// `qso` table (see `QSORepository.countForCallsign`), which is the single source
/// of truth and stays correct across edits, deletes, and imports automatically.
struct CallsignHistory: Codable, Identifiable, Equatable {
    var callsign: String
    var name: String?
    var qth: String?
    var grid: String?
    var lastWorked: Date?

    var id: String { callsign }

    init(
        callsign: String,
        name: String? = nil,
        qth: String? = nil,
        grid: String? = nil,
        lastWorked: Date? = nil
    ) {
        self.callsign = callsign
        self.name = name
        self.qth = qth
        self.grid = grid
        self.lastWorked = lastWorked
    }
}

extension CallsignHistory: FetchableRecord, PersistableRecord {
    static var databaseTableName = "callsignHistory"
}
